### PR TITLE
First SPV1 support

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for Bisoromi's SPV1 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/spv1/spv1.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(LED_CAPS, led_state.caps_lock);
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/spv1/spv1.c:22]: It seems you were bamboozled by the stealth LEDs

The keyboard has the three indicators plus a lot more sprinkled around, one on caps lock, another on scroll, two on the num locks

File: keyboards/spv1/spv1.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        writePin(LED_IND8, led_state.num_lock);
+    }
+    return res;
+}
+
+layer_state_t layer_state_set_kb(layer_state_t state) {
+    if (get_highest_layer(state) > 0) writePin(LED_IND7, 1);
+    else writePin(LED_IND7, 0);
+    return state;
+}
```
Comment (drashna) [inline keyboards/spv1/spv1.c:40]: no init code? 

```suggestion
void led_update_ports(led_t led_state) {
    writePin(LED_CAPS, led_state.caps_lock);
    writePin(LED_IND1, led_state.caps_lock);
    writePin(LED_IND2, led_state.num_lock);
    writePin(LED_IND3, led_state.scroll_lock);

    writePin(LED_IND4, led_state.scroll_lock);
    writePin(LED_IND6, led_state.scroll_lock);

    writePin(LED_IND5, led_state.num_lock);
    writePin(LED_IND8, led_state.num_lock);
}

layer_state_t layer_state_set_kb(layer_state_t state) {
    if (get_highest_layer(state) > 0) writePin(LED_IND7, 1);
    else writePin(LED_IND7, 0);
    return layer_state_set_user(state);
}
```

File: keyboards/spv1/readme.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# SPV1 keyboard QMK firmware
+
+![spv1](https://i.imgur.com/C2tNDzY.jpg)
```
Comment (dunk2k) [inline keyboards/spv1/readme.md:3]:
```suggestion
![spv1](https://i.imgur.com/C2tNDzYh.jpg)
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/spv1/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+BACKLIGHT_DRIVER = pwm
```
Comment (dunk2k) [inline keyboards/spv1/rules.mk:1]: migrate to `info.json`
```suggestion
# File is intentionally blank
```

File: keyboards/spv1/info.json [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        "rows": ["C3" ,"C2" ,"C1" ,"C0","A15","A7" ,"A0" ]
+    },
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "backlight": true
+    },
+    "backlight": {
+        "breathing": true,
```
Comment (dunk2k) [inline keyboards/spv1/info.json:24]: migrated from `rules.mk`
```suggestion
        "driver": "pwm",
        "breathing": true,
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/spv1/info.json (hunk 5 above)
Comment (Gondolindrim) [reply keyboards/spv1/info.json:24]: Done in 23530fcafc

File: keyboards/spv1/rules.mk (hunk 4 above)
Comment (Gondolindrim) [reply keyboards/spv1/rules.mk:1]: Done in 23530fcafc

File: keyboards/spv1/readme.md (hunk 3 above)
Comment (Gondolindrim) [reply keyboards/spv1/readme.md:3]: Done in 60c7b8b7e5

File: keyboards/spv1/spv1.c (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/spv1/spv1.c:40]: Code refactored and pin inits added in 2355a6d354 .

File: keyboards/spv1/spv1.c [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+void keyboard_post_init_kb(void) {
+     setPinOutput(LED_CAPS);
+     setPinOutput(LED_IND1);
+     setPinOutput(LED_IND2);
+     setPinOutput(LED_IND3);
+     setPinOutput(LED_IND4);
+     setPinOutput(LED_IND5);
+     setPinOutput(LED_IND6);
+     setPinOutput(LED_IND7);
+     setPinOutput(LED_IND8);
```
Comment (waffle87) [inline keyboards/spv1/spv1.c:28]:
```suggestion
     setPinOutput(LED_IND8);
     keyboard_post_init_user();
```

File: keyboards/spv1/info.json [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+                {"label":"F4",          "matrix":[ 0, 8], "x":8.75 , "y":0               }, 
+                {"label":"F5",          "matrix":[ 0, 9], "x":10.25, "y":0               }, 
+                {"label":"F6",          "matrix":[ 0,10], "x":11.25, "y":0               }, 
+                {"label":"F7",          "matrix":[ 0,11], "x":12.25, "y":0               }, 
+                {"label":"F8",          "matrix":[ 0,12], "x":13.25, "y":0               }, 
+                {"label":"F9",          "matrix":[ 0,13], "x":14.75, "y":0               }, 
+                {"label":"F10",         "matrix":[ 0,14], "x":15.75, "y":0               }, 
+                {"label":"F11",         "matrix":[ 0,15], "x":16.75, "y":0               }, 
+                {"label":"F12",         "matrix":[ 0,16], "x":17.75, "y":0               }, 
+                {                       "matrix":[ 0,17], "x":17.25, "y":1.25            },
```
[the remark this answers is not in the document]
Comment (Gondolindrim) [reply keyboards/spv1/info.json]: Done in 64b5dc7d58

File: keyboards/spv1/config.h [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+#define LED_IND2 B15
+#define LED_IND3 C6
+#define LED_IND4 B13
+#define LED_IND5 B1
+#define LED_IND6 B12
+#define LED_IND7 B10
+#define LED_IND8 B0
+
+#define WEAR_LEVELING_LOGICAL_SIZE 2048
+#define WEAR_LEVELING_BACKING_SIZE 4096
```
Comment (zvecr) [inline keyboards/spv1/config.h:34]:
```suggestion
```

File: keyboards/spv1/info.json [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+        "mousekey": true,
+        "backlight": true
+    },
+    "backlight": {
+        "driver": "pwm",
+        "breathing": true,
+        "levels": 30,
+        "pin": "C7"
+    },
+   "diode_direction": "COL2ROW",
```
Comment (zvecr) [inline keyboards/spv1/info.json:29]:
```suggestion
    },
    "eeprom": {
        "wear_leveling": {
            "backing_size": 4096
        }
    },
    "diode_direction": "COL2ROW",
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/spv1/info.json
Comment (fauxpark) [inline keyboards/spv1/info.json:1]: Please run `qmk format-json -i` on this file after all other review suggestions.

**Pull request closed without merging**